packages/devtools_app/lib/src/survey.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:convert';
+
+import 'dart:io' if (dart.library.html) 'dart:html';
+
+import 'package:flutter/widgets.dart';
+import 'package:http/http.dart';
+
+import 'config_specific/logger/logger.dart';
+import 'config_specific/server/server.dart' as server;
+import 'notifications.dart';
+import 'utils.dart';
+
+class SurveyService {
+  static const noThanksLabel = 'NO THANKS';
+
+  static const takeSurveyLabel = 'TAKE SURVEY';
+
+  static const maxShowSurveyCount = 5;
+
+  static const metadataUrl =
+      'https://flutter.dev/f/dart-devtools-survey-metadata.json';
+
+  /// Duration for which we should show the survey notification.
+  ///
+  /// We use a very long time here to give the appearance of a persistent
+  /// notification. The user will need to interact with the prompt to dismiss
+  /// it.
+  static const notificationDuration = Duration(days: 1);
+
+  /// The amount of time (one day) to wait between checks to the file at
+  /// [metadataUrl].
+  static const checkInterval = Duration(days: 1);
+
+  DevToolsSurvey _cachedSurvey;
+
+  Future<DevToolsSurvey> get activeSurvey async {
+    if (await shouldFetchSurveyContent()) {
+      _cachedSurvey = await fetchSurveyContent();
+      if (_cachedSurvey != null) {
+        await server.setActiveSurvey(_cachedSurvey.id);
+      }
+    }
+
+    if (await shouldShowSurvey()) {
+      return _cachedSurvey;
+    }
+    return null;
+  }
+
+  void maybeShowSurveyPrompt(BuildContext context) async {
+    final survey = await activeSurvey;
+    if (survey != null) {
+      final message = survey.title;
+      final actions = [
+        NotificationAction(
+          noThanksLabel,
+          () => _noThanksPressed(message, context),
+        ),
+        NotificationAction(
+          takeSurveyLabel,
+          () => _takeSurveyPressed(survey.url, context),
+          isPrimary: true,
+        ),
+      ];
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        final didPush = Notifications.of(context).push(
+          message,
+          actions: actions,
+          duration: notificationDuration,
+          allowDuplicates: false,
+        );
+        if (didPush) {
+          server.incrementSurveyShownCount();
+        }
+      });
+    }
+  }
+
+  Future<bool> shouldShowSurvey() async {
+    if (_cachedSurvey == null) return false;
+
+    final surveyShownCount = await server.surveyShownCount();
+    if (surveyShownCount >= maxShowSurveyCount) return false;
+
+    final surveyActionTaken = await server.surveyActionTaken();
+    if (surveyActionTaken) return false;
+
+    final currentTimeMs = DateTime.now().millisecondsSinceEpoch;
+    final activeSurveyRange = Range(
+      _cachedSurvey.startDate.millisecondsSinceEpoch,
+      _cachedSurvey.endDate.millisecondsSinceEpoch,
+    );
+    return activeSurveyRange.contains(currentTimeMs);
+  }
+
+  Future<bool> shouldFetchSurveyContent() async {
```

Comment:
lets remove this method and just rely on browser caching. loading one tiny json file from a server on devtools startup is completely fine if we don't block devtools startup to do it. if we are concerned we can tweak caching settings on the flutter.dev website to keep it cached for a day.